### DISCUSSION TITLE
[Projects Beta] issue history not updated when moved issue between columns or change status (Board)

### DISCUSSION THREAD

**Discussion by Marshevskyy** [community/community#5859] [Projects and Issues] [answered]
Hi. I'm excited to try a new Beta version, we already moved some of our projects to Beta.

One issue I found is related to history.
Issue history not updated when moved issue between columns (Board) or change status.
In old projects, any changes (including status) were added to the history.

Adding screenshot.
<img width="1275" alt="Screenshot 2021-09-28 at 17 43 15" src="https://user-images.githubusercontent.com/6871431/135110428-f6a2cde5-4168-4695-a3bf-ca87bc9a7547.png">

Please let me know if you need more details.

EDITED: there is no history about who and when added issues to project

**Comment by zalesky** [comment]
this issue bothers me as well

**Comment by levikoenig** [comment]
Upvoted this one too! Seeing the history of when an issue changed from previous statuses is highly valuable. I need to see who/when changed the status. I have no idea who to ask if I don't know why an issue is in a certain status. 

 This is available in the current 'Projects' just not in the Beta.

**Reply by VMois** [reply]
Totally agree.

**Comment by jldec** [comment]
Wondering if this is task 3 from https://github.com/github/roadmap/issues/287 ?
Q1 on [roadmap board](https://github.com/orgs/github/projects/4247/views/1?filterQuery=parity)

> - [ ] Project events are included in issue / pr timelines

**Reply by levikoenig** [reply]
Looks likely! Fingers crossed.

**Reply by tayloramurphy** [reply]
Seems like https://github.com/github/roadmap/issues/483 may actually be the issue. Looking forward to this as not tracking changes is worrisome for me!

**Reply by tpak** [reply]
@ankneis removed this **"Project events are included in issue / pr timelines"** from the roadmap item [483](https://github.com/github/roadmap/issues/287) yesterday - I can not find any explanation or other issue that is taking its place 

@ankneis - would you please be kind enough to comment or point us to a new roadmap item? This is a heavily relied on feature by many of us and just seeing it disappear is frustrating :-(

(follow up: apologies calling you out, not sure who else to ask and we all had our hopes up)

**Reply by tayloramurphy** [reply]
@tpak thanks for highlighting that. Disappointing to see it was removed. I'll echo that having somewhere to follow updates on this feature are needed. Not having this history makes Projects very fragile for work planning and trusting nothing is shifting under my feet!

**Reply by ankneis** [reply]
👋  - Looks like this [comment](https://github.com/github/roadmap-internal/issues/503#issuecomment-1092876610) mentioned "Project events in issue/pr timelines" was a "ways out". I'd follow up with @timoyates and @azenmatt to get a more concrete timeline since it is also not listed on the [Milestone 2 Issue](https://github.com/github/roadmap-internal/issues/665). 

cc @tpak

**Reply by azenMatt** [reply]
We are working on this at the moment 👍 there is a bit more complexity to this one than usual events so it has taken a bit longer, but it will be with you shortly. There is a roadmap issue waiting to be published that puts this in the `Oct-Dev` release segment.

**Reply by tpak** [reply]
@timoyates @ankneis will someone be keeping us up to date while @azenMatt is on leave for the next several months? Seems like this one is due here in Q4 2022 but still not showing on the roadmap as @azenMatt has suggested recently.

**Reply by ankneis** [reply]
@tpak There is an internal roadmap issue that is currently being reviewed. It has not yet been published because we are publishing new issues from the internal roadmap to the public roadmap on October 19.

**Reply by tpak** [reply]
@ankneis, thank you for the reply and shedding a little light on your process, unfortunately, you are linking to the internal or private repo, so again, we get a 404 

I guess we will have to wait until 19-October and hope that we get an update. As I said here: https://github.com/orgs/community/discussions/5859#discussioncomment-3665175 communications on this is key and this functionality is all about communications.

**Reply by ankneis** [reply]
@tpak Sorry for the confusion and linking to a repo that you cannot access. Summary here is to look out for the published and updated issue on October 19.

**Reply by 64J0** [reply]
Hello people, it's November 21 already, and I don't see this feature in my issues yet. Any update about this demand?

**Comment by ian-steffy** [comment]
This is a pretty serious issue and I question the logic behind the priorities of their roadmap. Why would a task like adding 'Themes' to github projects beta be before a task like this? 

They need to really double check which features are nice-to-have and which features, like this one, affect the core functionality of any project management tool. Giving Github Projects Beta the core functionality that already exists in Github Projects should be priority 1.

**Comment by tpak** [comment]
@ankneis thanks for the reply 🙏 --  I guess since you removed it I was hoping you
could shed light on why/when we will see parity with classic projects. Also
the link you provided 404’s

This item provides critical communication and info about the journey an
issue takes within a project or projects. Who moved an issue around and
when often provides a critical bit of info to devs and other actors.  I
find it odd that classic projects is now deprecated and such glaring
functionality left out of the new product.

@timoyates? @azenMatt is there any update or info you can provide?
(although, based on @azenMatt not commenting or updating other issues such
as issue templates in 3 months it’s not clear he is still around?)

Communication is key, this issue is all about communication. Both in the
real and meta sense :-)

**Reply by azenMatt** [reply]
replied in the thread [here](https://github.com/orgs/community/discussions/5859#discussioncomment-3674135) 😄

**Comment by tpak** [comment]
@timoyates @ankneis can someone please give us an update on this? It has not appeared as suggested as item on the roadmap -  @azenMatt has teased that this critical functionality was pending and would be available shortly before he went on extended leave and when @ankneis had removed it from the parity issue as a bullet point https://github.com/github/roadmap/issues/483 

We will not move our projects until we can see the activity in the issues; even in my tiny organization, it is information we rely on.

**Reply by timoyates** [reply]
@tpak Thanks for the ping! We are still working on this feature. It was removed from the referenced issue as planned but we missed the cut on adding it to another. It will re-appear in the next roadmap update!

**Reply by tpak** [reply]
@timoyates can you provide any guidance on "when" we will see it - this is a pretty basic requirement from our POV - to be able to see who/what/when issues are added/moved/removed from a project and into what status/column and by whom - very surprising it is not a priority to get this essential functionality in a project management tool

**Reply by tpak** [reply]
Please understand, it is very frustrating to get dribbles of information - we'd like to move over to new project but this is absolutely a requirement and a blocker for us

**Reply by 64J0** [reply]
+1 for @tpak. It really seems an easy feature to be added IMO, I don't understand why it's taking too long. Can you provide some deadline for this feature to be added?

**Reply by timoyates** [reply]
Hi folks, this feature was re-added to the roadmap two weeks ago. The issue to subscribe to is the following: https://github.com/github/roadmap/issues/613

Rest assured - development is active and you will be the first to know when it is rolled out or we kick off early access.

**Comment by 64J0** [comment]
+1, this feature would be very useful!

**Comment by shkkmo** [comment]
I need someway to track who changes the status of an issue. Even if this doesn't show up on the history of the issue itself (which it should), there should atleast be a log in the project somewhere of all status changes, who they were done by and when they happened..

**Reply by fmagarrotebridge** [reply]
Eu concordo

**Comment by azenMatt** [comment, marked as answer]
Hi everyone - we just released an update that now includes this history! https://github.blog/changelog/2023-01-12-github-issues-january-12th-update/

This means there are events for: 

- Adding to a project
- Removing (deleting) from a project 
- Changing the status value of an item in a project 

This works for both issues and Pull Requests. 

We ran all the existing events into the system for this so all events you've previously performed are included as well as any new events. 

Please give it a try and let me know if you have any thoughts / questions 😄

**Reply by slinna** [reply]
This is great news @azenMatt! 

Are the status changes also available via the GraphQL API? If not, is there ETA to share?

**Reply by joe-woolvett** [reply]
Thank you!

**Reply by 64J0** [reply]
Nice! Thanks for adding this feature!

**Reply by azenMatt** [reply]
> Are the status changes also available via the GraphQL API? If not, is there ETA to share?

No ETA but we are currently reviewing the next steps for our improved `timeline service` (of which this is the first release) and this is a top candidate. I'l share with the team that this is a request 😄

**Reply by tpak** [reply]
Welcome back @azenMatt, and thank you, this is great news!

**Reply by tayloramurphy** [reply]
This is great! As with any new feature I'm sure folks have a follow-on - are there plans to reflex custom field changes in the history? If I make a weight field for a project it can change without any history recorded. Happy to follow an issue / discussion for this as well. Thanks @azenMatt

**Reply by azenMatt** [reply]
Hey @tayloramurphy - our thinking was that this might make the timeline in the issue/pr a bit too messy. However - this information is certainly needed. Ideally this would come in an 'activity history' style page inside the project. We do have this tracked in our public roadmap here - https://github.com/orgs/github/projects/4247/views/7?pane=issue&itemId=14542755 - although I think that date is a bit optimistic. I'l talk to the team and get it updated if it is.

**Reply by tayloramurphy** [reply]
@azenMatt really appreciate the response and the link! Agreed that it would be noisy - a way that GitLab solved this was to have a toggle on the issue page to view comments, activity, or both. That would help a bit. https://docs.gitlab.com/ee/user/discussions/#show-only-comments

I should mention too - I actually already have this b/c of the [Refined GitHub extension](https://chrome.google.com/webstore/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?hl=en)

<img width="301" alt="Screenshot 2023-01-20 at 10 06 23 AM" src="https://user-images.githubusercontent.com/5394188/213747148-fce425f1-c4d8-4bbd-af9c-abb5c4f269d5.png">

**Comment by tpak** [comment]
Hi, @azenMatt and @ankneis. Thanks for delivering this! I have been so swamped that I have had little to no time to give this a proper test, but I like what I see so far!
 
I just wanted to drop in and provide a (very) little feedback, say thanks, and let you know you and your teams are appreciated! 🙏

**Comment by galah92** [comment]
Hi @azenMatt & @ankneis,

I see history of field changes when these fields are Single Select (like status) field, but do not see changes of other field types (mostly Iteration, but also Number).
Is this something that should be supported?

**Reply by tayloramurphy** [reply]
@galah92 the other threads have an answer to your question https://github.com/orgs/community/discussions/5859#discussioncomment-4735835

**Reply by galah92** [reply]
Oh thanks. @azenMatt says there it's begin tracked but the link he added shows the issue is completed, though.

**Reply by azenMatt** [reply]
it looks like it was moved out of that release. We are currently working with the team to the issues tab in that project updated which should give more clarity cc @2percentsilk

**Reply by galah92** [reply]
@azenMatt got it, still waiting for some clarity on this, thanks. It's what's blocking me from adding editing permissions to the PMs in my org, I need visibility as to who changed what.

**Reply by Alex-ley-scrub** [reply]
I also don't see when the Priority field changes which is a single select just like the Status field?

**Reply by daniel-evers** [reply]
Unfortunately, I cannot find any clarifying information on this point. @azenMatt Are there still plans to provide a history in the issues timeline for fields other than the status? I would love to see this feature.

**Reply by ShaneOH** [reply]
@azenMatt Any updates on this 2 years on? Having the timeline for status changes is nice, but we also need visibility on other field changes, e.g. it's important to know who is moving issues between sprints (iterative fields) and when this was done, but we have no visibility on this. We _did_ have visibility on this in the _old_ Projects system so this has been a pretty big downgrade for us.

**Reply by azenMatt** [reply]
Sorry no updates but i know this is painful. I don't have a timeline but we would love to get this fixed

**Reply by ShaneOH** [reply]
Appreciate the response & acknowledgement @azenMatt ! Hopefully feedback on this can continue to be passed on -- expediting the fix in some way would be great.

**Comment by shadeMe** [comment]
Thanks for porting this feature from classic Projects! Has this feature been rolled out to all users? We recently migrated our classic board to the beta, but PRs in our [organization](https://github.com/explosion)'s repositories still do not show status changes in their timeline.

**Comment by MarioGith** [comment]
Hi everyone,
I restart the conversation about the accessibility of Issue history inside the GraphQL API. Is there an ETA on it ? 
I'm a DevOps laboratory manager at school and I need this to be accessible for my courses 😄
Thank you !

**Reply by Galadirith** [reply]
@MarioGith if this is still important for you I would suggest upvoting https://github.com/orgs/community/discussions/49602 which is currently [the highest ranked _API and Webhooks_ discussion](https://github.com/orgs/community/discussions/categories/api-and-webhooks?discussions_q=is%3Aopen+category%3A%22API+and+Webhooks%22+sort%3Atop) that is related to this missing feature of the GraphQL API. @jesuslinares has also summarised the other related discussion in [their comment](https://github.com/orgs/community/discussions/49602#discussioncomment-6108733).

**Comment by vadimfrolov** [comment]
Made JS scrapper for this problem 

https://github.com/vadimfrolov/Github-Issue-parser-JS

**Comment by slinna** [comment]
This is still very much needed fix to the GraphQL API. What's the point of supporting this data on legacy projects but not with the new ones? Please provide us issue history including the information of when an issue moved between columns or changed status.

**Comment by Ygilany** [comment]
This is not resolved on GHES!

**Reply by Ygilany** [reply]
@azenMatt & @ankneis Is there an ETA for this

**Reply by azenMatt** [reply]
Hi @Ygilany - acknowledged this is still a gap on the GHES side. There is a small chance it may make 3.16, but 3.17 is much more likely.

**Reply by gravelpot** [reply]
@azenMatt It doesn't appear that this feature has landed as of GHES 3.17.1. Any updated ETA?

**Reply by azenMatt** [reply]
Thank you for pointing this out for us @gravelpot - I apologise this was an oversight. It will now be in the 3.18 and the next version of 3.17.
